service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -326,6 +326,7 @@ protected synchronized void cleanupOperationLog(final long operationLogCleanupDe
         if (operationLogCleanupDelayMs > 0) {
           scheduledExecutorService.schedule(new OperationLogCleaner(operationLog), operationLogCleanupDelayMs,
             TimeUnit.MILLISECONDS);
+          scheduledExecutorService.shutdown();
```

Comment:
If I understand correctly, then the pool is created regardless of the config. We might want to close outside of the conditional expressions. What do you think?

Reply:
Or even beter, move the executor creation to here too....

Reply:
Yes, thanks very much for the advice.